src/shirt/string_render.clj

```diff
@@ -52,8 +52,10 @@
 
 (def num-candidates 2500)
 (defn best-partitions [s ^Graphics g width y height]
-  (let [candidates (for [i (range num-candidates)] (candidate-partition s y height width i g))]
-    (first (sort-by (partial candidate-badness g width height) candidates))))
```

Comment:
we probably don't want `memoize` as candidate-badness gets called a lot, with a lot of different parameters, and I don't know how Graphics work but they might be large and if we memoize we can't GC them. Probably